### PR TITLE
Consider devices in tasksUpdated

### LINKED ISSUE #6644: Modifying device count doesn't cause scheduling
### Nomad version
Nomad 0.9.0, 0.9.6, 0.10.1

### Operating system and Environment details
MacOS, example device plugin

### Issue
After updating the device count in the `resources` stanza, the scheduler does not detect the need for a destructive allocation update, and the updated device count is not reflected in the updated allocation.

The reason is, at least, that the task update check does not consider the new device field: 
https://github.com/hashicorp/nomad/blob/v0.10.1/scheduler/util.go#L399-L404

### Reproduction steps
1. Run job and note allocation resources:
> Device Stats
> hashicorp/example/hello[hello-1]  1
2. Modify job to increase device "example" count, note that allocation is updated: 
>     Allocation "0abefd49" modified: node "a1a38f1b", group "cache"
3. Examine allocation resources, no new device was attached.

### Job file (if appropriate)
```hcl
job "example" {
  datacenters = ["dc1"]
  group "cache" {
    task "redis" {
      driver = "docker"
      config {
        image = "redis:3.2"
      }
      resources {
        device "example" {
          count = 1
        }
      }
    }
  }
}
```

### Nomad Server logs (if appropriate)
On job update:
```
    2019-11-07T16:47:30.580Z [DEBUG] worker.service_sched: reconciled current state with desired state: eval_id=aba15a40-2bb7-74f7-8b47-651f81445050 job_id=example namespace=default results="Total changes: (place 0) (destructive 0) (inplace 1) (stop 0)
Desired Changes for "cache": (place 0) (inplace 1) (destructive 0) (stop 0) (migrate 0) (ignore 0) (canary 0)"
```

### PR DESCRIPTION
**Pull request by cgbaker**
the scheduler checks whether task changes require a restart, this needed to be updated to consider devices

resolves #6644

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,7 @@ IMPROVEMENTS:
 BUG FIXES:
 
  * cli: Make scoring column orders consistent `nomad alloc status` [[GH-6609](https://github.com/hashicorp/nomad/issues/6609)]
+ * server: Changes to devices in resource stanza should cause rescheduling [[GH-6644](https://github.com/hashicorp/nomad/issues/6644)]
```
Comment (schmichael) [inline CHANGELOG.md]:
```suggestion
 * scheduler: Changes to devices in resource stanza should cause rescheduling [[GH-6644](https://github.com/hashicorp/nomad/issues/6644)]
```

File: scheduler/util_test.go [hunk 2]
```diff
@@ -527,6 +527,24 @@ func TestTasksUpdated(t *testing.T) {
...
+		},
+	}
+	j11d2 := mock.Job()
+	j11d2.TaskGroups[0].Tasks[0].Resources.Devices = structs.ResourceDevices{
+		&structs.RequestedDevice{
+			Name:  "gpu",
+			Count: 2,
+		},
+	}
+	if !tasksUpdated(j11d1, j11d2, name) {
```
Comment (schmichael) [inline scheduler/util_test.go]:
```go
require.True(t, tasksUpdated(...))
```

File: CHANGELOG.md (hunk 1 above)
Comment (cgbaker) [reply CHANGELOG.md]: funny, i wrote `scheduler` first, then changed it to `server`

File: scheduler/util_test.go (hunk 2 above)
Comment (cgbaker) [reply scheduler/util_test.go]: @schmichael , i couldn't see changing one instance to `require`, so i changed this whole test suite.

**Pull request merged**